### ISSUE TITLE
bootkube start attempts to create the assets too fast

### ISSUE THREAD

**Issue by abourget** [kubernetes-incubator/bootkube#358]
In many deployments I've made, I see:

```
[restful] 2017/03/08 21:52:11 log.go:30: [restful/swagger] listing is available at https://10.95.123.237:443/swaggerapi/
[restful] 2017/03/08 21:52:11 log.go:30: [restful/swagger] https://10.95.123.237:443/swaggerui/ is mapped to folder /swagger-ui/
I0308 21:52:12.035715    8129 serve.go:88] Serving securely on 0.0.0.0:443
I0308 21:52:12.035861    8129 serve.go:102] Serving insecurely on 127.0.0.1:8080
Creating self-hosted assets...
I0308 21:52:32.005063    8129 trace.go:61] Trace "Create /api/v1/namespaces/kube-system/secrets" (started 2017-03-08 21:52:22.004196352 +0000 UTC):
[41.031µs] [41.031µs] About to convert to expected version
[257.444µs] [216.413µs] Conversion done
[10.000828166s] [10.000570722s] END
```

The important bit here is `Creating self-hosted assets`.. it starts attemptings to create the assets before the cluster is *really* ready.. Often, the cluster becomes ready (some times.. it's only the default namespace is missing), but then creating the assets fail.. and we need to start all over.

The trigger to `Creating self-hosted assets`, ideally, would poke a bit more to know the internal state, readiness of etcd or whatnot.

**Comment by aaronlevy** [comment]
Related: https://github.com/kubernetes-incubator/bootkube/issues/319

Maybe waiting on the kube-system namespace would be sufficient?

**Comment by aaronlevy** [comment]
I'm going to close this as duplicate of https://github.com/kubernetes-incubator/bootkube/issues/319 - as I think if we waited until kube-system namespace existed - that might be a bit more safe and should resolve this issue.
